### PR TITLE
chore(gear): replace dust handling entirely with gear-bank pallet

### PR DESCRIPTION
**Pull request by playX18**
Removes dust handling, instead use gear-bank and UnusedValue from it.

### CODE REVIEW

File: pallets/gear/src/internal.rs [hunk 1]
```diff
@@ -1078,28 +1074,16 @@ where
...
-                            value,
-                        ));
-                    }
-                    e => {
-                        // Other errors are ruled out by the protocol guarantees.
-                        let err_msg = format!(
-                            "send_user_message: failed to transfer value. Got error: {e:?}"
-                        );
+            GearBank::<T>::deposit_value(&from, value, false)
+                .and_then(|_| GearBank::<T>::transfer_value(&from, &to, value))
```
Comment (techraed) [inline pallets/gear/src/internal.rs:1078]: @playX18 
the branch above, btw, performs `deposit_value`, so just move the `deposit_value` out of both `if-else` branches and just do `transfer_value` here.

File: pallets/gear/src/internal.rs [hunk 2]
```diff
@@ -1078,28 +1074,16 @@ where
...
-                        <CurrencyOf<T> as fungible::Unbalanced<_>>::handle_dust(fungible::Dust(
-                            value,
-                        ));
-                    }
-                    e => {
-                        // Other errors are ruled out by the protocol guarantees.
-                        let err_msg = format!(
-                            "send_user_message: failed to transfer value. Got error: {e:?}"
-                        );
+            GearBank::<T>::deposit_value(&from, value, false)
```
[the remark this answers is not in the document]
Comment (techraed) [reply pallets/gear/src/internal.rs:1077]: Can't really say anything about these changes being more "costly", but it's 1) neat, 2) easier to understand, 3) exposes less details of the `Currency`.

what do you think?

File: pallets/gear/src/internal.rs [hunk 3]
```diff
@@ -1078,27 +1073,14 @@ where
             // Note that we have no guarantees of the user account to exist. Since no minimum
             // transfer value is enforced, the transfer can fail. Handle it gracefully.
             // TODO #4018 Introduce a safer way to handle this.
```
Comment (breathx) [inline pallets/gear/src/internal.rs:1075]: todo solved?

File: pallets/gear/src/tests.rs [hunk 4]
```diff
@@ -15711,7 +15709,7 @@ fn dust_in_message_to_user_handled_ok() {
 
         // USER_1 account doesn't receive the funds; instead, the dust handler kicks in.
         assert_eq!(CurrencyOf::<Test>::free_balance(USER_1), 0);
-        assert_eq!(CurrencyOf::<Test>::free_balance(DUST_TRAP_TARGET), ed + 300);
+        assert_eq!(pallet_gear_bank::UnusedValue::<Test>::get(), 300);
```
Comment (breathx) [inline pallets/gear/src/tests.rs:15712]: add some comments pls for each change in tests

**Review by breathx — APPROVED** [review]
@techraed pls re-review

**Review by techraed — APPROVED** [review]

**Comment by breathx** [comment]
pls action comments above and ready for merging

**Pull request merged**